src/Symfony/Bundle/SecurityBundle/Resources/config/security.php

```diff
@@ -0,0 +1,276 @@
+            ->args([
+                service('security.expression_language'),
+                service('security.authentication.trust_resolver'),
+                service('security.authorization_checker'),
+                service('security.role_hierarchy')->nullOnInvalid()
+            ])
+            ->tag('security.voter', ['priority' => 245])
+
+        // Firewall related services
+        ->set('security.firewall', FirewallListener::class)
+            ->tag('kernel.event_subscriber')
+            ->args([
+                service('security.firewall.map'),
+                service('event_dispatcher'),
+                service('security.logout_url_generator'),
+            ])
+        ->alias(Firewall::class, 'security.firewall')
+
+        ->set('security.firewall.map', FirewallMap::class)
+            ->args([
+                abstract_arg('Firewall context locator'),
+                abstract_arg('Request matchers'),
+            ])
+
+        ->set('security.firewall.context', FirewallContext::class)
+            ->abstract()
+            ->args([
+                [],
+                service('security.exception_listener'),
+                abstract_arg('LogoutListener'),
+                abstract_arg('FirewallConfig'),
+            ])
+
+        ->set('security.firewall.lazy_context', LazyFirewallContext::class)
+            ->abstract()
+            ->args([
+                [],
+                service('security.exception_listener'),
+                abstract_arg('LogoutListener'),
+                abstract_arg('FirewallConfig'),
+                service('security.untracked_token_storage'),
+            ])
+
+        ->set('security.firewall.config', FirewallConfig::class)
+            ->abstract()
+            ->args([
+                '',
+                abstract_arg('user_checker'),
+                abstract_arg('request_matcher'),
+                false, // security enabled
+                false, // stateless
+                null,
+                null,
+                null,
+                null,
+                null,
+                [], // listeners
+                null, // switch_user
+            ])
+
+        ->set('security.logout_url_generator', LogoutUrlGenerator::class)
+            ->args([
+                service('request_stack')->nullOnInvalid(),
+                service('router')->nullOnInvalid(),
+                service('security.token_storage')->nullOnInvalid(),
+            ])
+
+        // Provisioning
+        ->set('security.user.provider.missing', MissingUserProvider::class)
+            ->abstract()
+            ->args([
+                abstract_arg('firewall')
+            ])
+
+        ->set('security.user.provider.in_memory', InMemoryUserProvider::class)
+            ->abstract()
+
+        ->set('security.user.provider.ldap', LdapUserProvider::class)
+            ->abstract()
+            ->args([
+                abstract_arg('firewall'),
```

Comment:
This one should be removed

Reply:
removed ??

Reply:
Yes there is only 9 args in XML file :)
Firewall was not in the xml file